pure_pursuit_wrapper/src/pure_pursuit_wrapper.cpp

```diff
@@ -14,42 +14,159 @@
+  config_.speed_to_lookahead_ratio = declare_parameter<double>("speed_to_lookahead_ratio", config_.speed_to_lookahead_ratio);
+  config_.is_interpolate_lookahead_point = declare_parameter<bool>("is_interpolate_lookahead_point", config_.is_interpolate_lookahead_point);
+  config_.is_delay_compensation = declare_parameter<bool>("is_delay_compensation", config_.is_delay_compensation);
+  config_.emergency_stop_distance = declare_parameter<double>("emergency_stop_distance", config_.emergency_stop_distance);
+  config_.speed_thres_traveling_direction = declare_parameter<double>("speed_thres_traveling_direction", config_.speed_thres_traveling_direction);
+  config_.dist_front_rear_wheels = declare_parameter<double>("dist_front_rear_wheels", config_.dist_front_rear_wheels);
+}
+
+carma_ros2_utils::CallbackReturn PurePursuitWrapperNode::on_configure_plugin()
+{
+  config_ = PurePursuitWrapperConfig();
+  get_parameter<double>("vehicle_response_lag", config_.vehicle_response_lag);
+  get_parameter<double>("minimum_lookahead_distance", config_.minimum_lookahead_distance);
+  get_parameter<double>("maximum_lookahead_distance", config_.maximum_lookahead_distance);
+  get_parameter<double>("speed_to_lookahead_ratio", config_.speed_to_lookahead_ratio);
+  get_parameter<bool>("is_interpolate_lookahead_point", config_.is_interpolate_lookahead_point);
+  get_parameter<bool>("is_delay_compensation", config_.is_delay_compensation);
+  get_parameter<double>("emergency_stop_distance", config_.emergency_stop_distance);
+  get_parameter<double>("speed_thres_traveling_direction", config_.speed_thres_traveling_direction);
+  get_parameter<double>("dist_front_rear_wheels", config_.dist_front_rear_wheels);
+
+  RCLCPP_INFO_STREAM(rclcpp::get_logger("pure_pursuit_wrapper"), "Loaded Params: " << config_);
+
+  // Register runtime parameter update callback
+  add_on_set_parameters_callback(std::bind(&PurePursuitWrapperNode::parameter_update_callback, this, std_ph::_1));
+  
+  // create config for pure_pursuit worker
+  pure_pursuit::Config cfg{
+    config_.minimum_lookahead_distance,
+    config_.maximum_lookahead_distance,
+    config_.speed_to_lookahead_ratio,
+    config_.is_interpolate_lookahead_point,
+    config_.is_delay_compensation,
+    config_.emergency_stop_distance,
+    config_.speed_thres_traveling_direction,
+    config_.dist_front_rear_wheels,
+  };
+  
+  pp_ = std::make_shared<pure_pursuit::PurePursuit>(cfg);
+
+  // Return success if everything initialized successfully
+  return CallbackReturn::SUCCESS;
+}
+
+motion::motion_common::State PurePursuitWrapperNode::convert_state(geometry_msgs::msg::PoseStamped pose, geometry_msgs::msg::TwistStamped twist)
+{
+  motion::motion_common::State state; //todo check if correct
+  state.header = pose.header;
+  state.state.x = pose.pose.position.x;
+  state.state.y = pose.pose.position.y;
+  state.state.z = pose.pose.position.z;
+  state.state.heading.real = pose.pose.orientation.w;
```

Comment:
I'm a bit skeptical of this 
It looks like the heading data type is autoware_auto_msgs/Complex32 https://github.com/usdot-fhwa-stol/autoware.auto/blob/8ce7d3f3ddab451a61ccea400232a4488df516be/autoware_auto_msgs/autoware_auto_msgs/msg/Complex32.idl

This has a real and imaginary piece and has references to quaternions and cos(yaw/2) in it. 

While "w" is the real component of the quaternion, I would expect the imaginary piece to also be needed. But I've actually never seen this representation before. It's almost like a 2d quaternion, but I've never seen such a thing in practice.

Reply:
added the imaginary part using pose.pose.orientation.z field which was the only non-zero field, and it worked!